frontend/src/metabase/setup/setup.unit.spec.tsx

```diff
@@ -0,0 +1,192 @@
+      });
+    });
+
+    describe("when selecting 'Embedding'", () => {
+      it("should hide the 'Add your data' step", async () => {
+        await setupForUsageQuestion();
+        fillUsageReason("embedding");
+        clickNextStep();
+
+        expect(screen.queryByText("Add your data")).not.toBeInTheDocument();
+
+        expect(getSection("Usage data preferences")).toHaveAttribute(
+          "aria-current",
+          "step",
+        );
+
+        expectSectionToHaveLabel("Usage data preferences", "4");
+      });
+    });
+
+    describe("when selecting 'A bit of both'", () => {
+      it("should keep the 'Add your data' step", async () => {
+        await setupForUsageQuestion();
+        fillUsageReason("both");
+        clickNextStep();
+
+        expect(screen.getByText("Add your data")).toBeInTheDocument();
+
+        expect(getSection("Add your data")).toHaveAttribute(
+          "aria-current",
+          "step",
+        );
+
+        expectSectionToHaveLabel("Add your data", "4");
+        expectSectionToHaveLabel("Usage data preferences", "5");
+      });
+    });
+
+    describe("when selecting 'Not sure yet'", () => {
+      it("should keep the 'Add your data' step", async () => {
+        await setupForUsageQuestion();
+        fillUsageReason("not-sure");
+        clickNextStep();
+
+        expect(screen.getByText("Add your data")).toBeInTheDocument();
+
+        expect(getSection("Add your data")).toHaveAttribute(
+          "aria-current",
+          "step",
+        );
+
+        expectSectionToHaveLabel("Add your data", "4");
+        expectSectionToHaveLabel("Usage data preferences", "5");
+      });
+    });
+  });
+});
+
+const getSection = (name: string) => screen.getByRole("listitem", { name });
+
+const clickNextStep = () =>
+  userEvent.click(screen.getByRole("button", { name: "Next" }));
+
+const skipWelcomeScreen = () =>
+  userEvent.click(screen.getByText("Let's get started"));
+
+const skipLanguageStep = () => clickNextStep();
+
+const submitUserInfoStep = async ({
+  firstName = "John",
+  lastName = "Smith",
+  email = "john@example.org",
+  companyName = "Acme",
+  password = "Monkeyabc123",
+} = {}) => {
+  fireEvent.change(screen.getByLabelText("First name"), {
+    target: { value: firstName },
+  });
+  fireEvent.change(screen.getByLabelText("Last name"), {
+    target: { value: lastName },
+  });
+  fireEvent.change(screen.getByLabelText("Email"), {
+    target: { value: email },
+  });
+  fireEvent.change(screen.getByLabelText("Company or team name"), {
+    target: { value: companyName },
+  });
+  fireEvent.change(screen.getByLabelText("Create a password"), {
+    target: { value: password },
+  });
+  fireEvent.change(screen.getByLabelText("Confirm your password"), {
+    target: { value: password },
+  });
+  await waitFor(() =>
+    expect(screen.getByRole("button", { name: "Next" })).toBeEnabled(),
+  );
+  clickNextStep();
+};
+
+const fillUsageReason = (usageReason: UsageReason) => {
```

Comment:
Technically, it's not a fill, but selecting a usage reason.

```suggestion
const selectUsageReason = (usageReason: UsageReason) => {
```

fill feels more like an input or a textbox where you need to type texts.